src/bitter/bit_field.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Steinwurf ApS 2016.
+// All Rights Reserved
+//
+// Distributed under the "BSD License". See the accompanying LICENSE.rst file.
+#pragma once
+
+#include "size_in_bits.hpp"
+
+#include <type_traits>
+#include <cstdint>
+#include <cassert>
+
+namespace bitter
+{
+template<typename DataType>
```

Comment:
Since there is no need for it to be a run-time parameter, if you pass it in as a template parameter you can do static_assert on it instead of normal assert